src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.PackageDependencyResolution.targets

```diff
@@ -32,8 +32,18 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ProjectAssetsCacheFile Condition="'$(ProjectAssetsCacheFile)' == ''">$(IntermediateOutputPath)$(MSBuildProjectName).assets.cache</ProjectAssetsCacheFile>
     <ProjectAssetsCacheFile>$([MSBuild]::NormalizePath($(MSBuildProjectDirectory), $(ProjectAssetsCacheFile)))</ProjectAssetsCacheFile>
 
-    <CopyLocalLockFileAssemblies Condition="'$(CopyLocalLockFileAssemblies)' == '' and '$(_TargetFrameworkVersionWithoutV)' >= '3.0'">true</CopyLocalLockFileAssemblies>
-    <CopyLocalLockFileAssemblies Condition="'$(CopyLocalLockFileAssemblies)' == ''">false</CopyLocalLockFileAssemblies>
+    <!-- Don't copy local for netstandard projects. -->
+    <CopyLocalLockFileAssemblies Condition="'$(CopyLocalLockFileAssemblies)' == '' and
+                                            '$(TargetFrameworkIdentifier)' == '.NETStandard'">false</CopyLocalLockFileAssemblies>
+
+    <!-- Don't copy local for netcoreapp projects before 3.0 or non-exe projects. -->
+    <CopyLocalLockFileAssemblies Condition="'$(CopyLocalLockFileAssemblies)' == '' and
+                                            '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                                            ('$(_TargetFrameworkVersionWithoutV)' &lt; '3.0' or
+                                             '$(_IsExecutable)' != 'true')">false</CopyLocalLockFileAssemblies>
```

Comment:
Should this use `HasRuntimeOutput` (which I think test projects set) rather than `_IsExecutable`?

Reply:
Fixed with the next push.